vtex-sites/gatsby.store
# chore: Single UI Provider

## What's the purpose of this pull request?
This PR joins duplicated contexts we had on our code. 

## How does it work?
Both `UIProvider` and `ModalProvider` had a goal of controlling the global UI state of our site. This created an interdependency between these contexts, generating a mess on our code. For example, for openning the cart on UIProvider we relied on functions provided by `ModalProvider`. Also, when openning the cart we had to invoke a function called `onModalOpen`, what makes no sense since the cart is a drawer, not a modal.

This PR addresses this issue by merging UIProvider and ModalProvider into a single context so we don't have that mess anymore. Also, I have changed the name of `onModalClose` and `onModalOpen` to `fadeOut` and `fadeIn` respectively. 

After these changes, the minicart toggle button went from:
```
const { closeCart } = useUI()
const { onModalClose } = useModal()

const onClick = () => {
    onModalClose()
    closeCart()
}

``` 

to: 
```
const { closeCart, fadeOut } = useUI()

const onClick = () => {
    fadeOut()
    closeCart()
}
```

## How to test it?
Make sure all fade effects are working correctly, as well as the scroll behavior when the modal is open
